### ISSUE TITLE
Email notification subject submission status not updated correctly

### ISSUE THREAD

**Issue by karthikkstar** [engram-design/Workflow#202]
### Describe the bug

Email notification subject submission status not updated correctly

**SUBJECT:** Your submission for "{{ submission.owner.title }}" has been  {{ submission.status }} 

The issue occurred only when the submission was approved or rejected the status was always "pending" on the following pages.
admin/workflow/submissions/edit/{submission_id}?site=australia ->On this page, when clicking "Save"
admin/workflow?site=australia&source=*&status=pending ->On this page, when clicking "set status"

![image](https://github.com/verbb/workflow/assets/85233749/70f1d9df-d79d-4664-8e88-5649200ddcd6)

Its working correctly on the entry page.
![image](https://github.com/verbb/workflow/assets/85233749/07b97256-63c6-4b99-a3bb-c03f33e53f70)

The email templates do not update the submission.status variable.

Thank you

### Steps to reproduce

1. Submit one workflow submission and approve or reject it from the submission listing or submission edit page. You will receive the wrong submission in the email notification. 

### Craft CMS version

4.6.1

### Plugin version

2.0.8

### Multi-site?

Yes

**Comment by engram-design** [comment]
Is an error thrown when you try to set the status via the submission index? I will note that you're not able to approve your own submissions if you're both the editor and the user toggling the status. That would miss the point of a review process.

**Comment by karthikkstar** [comment]
@engram-design Yes.On the submission index and submission edit page. We have different user groups for editors and publishers.
When you toggle the status:
1. If the submission status is "pending," I will modify it to either approve or reject. The email associated with the submission.status variable corresponds to the "Pending" status.
2. If the submission status is "reject," I will update it to approve. The email linked to the submission.status variable aligns with the "reject" status.

**Comment by engram-design** [comment]
Thanks for the clarification, I can't seem to reproduce that I'm afraid. Any time I toggle the submission state (from the submission index, or when editing a submission) the status is correctly updated, and any associated emails are using the status that I've just updated to.

Just to be clear in your scenario - is the status in the control panel still staying as pending (in your first example)? And it's just the email subject that seems to be "behind"?

Just to confirm things are working correctly, can you confirm if a new row in your `workflow_reviews` database table is created when you change the status? It should reflect the status you're setting the submission to.

For example, here's the reviews for a submission of mine 
<img width="606" alt="image" src="https://github.com/verbb/workflow/assets/1221575/8caad62c-d141-462b-9e6f-898ea2bca583">

The resulting email subject was `Your submission for "Test Entry" has been approved on Default Site.`

**Comment by karthikkstar** [comment]
@engram-design  Here is additional information for you to replicate.

**Email template :** 

**When a publisher approves or rejects an editor submission:**

**Subject:** Your submission for "{{ submission.owner.title }}" has been  {{ submission.status }} on the Website CMS.
**Body:** Hey {{ user.friendlyName }},

Your submission for {{ submission.owner.title }} has been {{ submission.status }} on the Website CMS.

**Email screenshot**
![image](https://github.com/verbb/workflow/assets/85233749/6d1816b1-eec7-4fc3-8dfa-660c4e93de27)

The submission's status is changed on the control panel during the updating, which is functioning well. Yes , it's just the email subject that seems to be "behind".


Yes. new row created on workflow_reviews table.
![image](https://github.com/verbb/workflow/assets/85233749/754e2f41-0524-4183-ba58-3f4b74d8722f)

**Comment by engram-design** [comment]
Oh, and the body content of the email would show the incorrect status too, right?

```
Your submission for test has been pending on the Website CMS.
```

Just in case it's a subject-specific issue, or any reference to status in the email.

**Comment by karthikkstar** [comment]
Yes. Both subject and body content show incorrect status. 

![image](https://github.com/verbb/workflow/assets/85233749/150c2e3d-c80a-4254-b5d5-cd62718ce3ee)

**Comment by karthikkstar** [comment]
@engram-design  another issue, I will adjust it to either approve or reject. The email linked to the {{submission.ownerCpUrl}} variable corresponds to a bad request.

![image](https://github.com/verbb/workflow/assets/85233749/796af7bb-58de-474f-9fae-de2730d5a1b3)

Thank you
